tests/trainer/test_trainer.py

```diff
@@ -1444,3 +1445,65 @@ def test_trainer_profiler_incorrect_arg_type(profiler):
                        match=r"Only None, bool, str and subclasses of `BaseProfiler`"
                              r" are valid values for `Trainer`'s `profiler` parameter. *"):
         Trainer(profiler=profiler)
+
+
+@pytest.mark.parametrize(
+    ["limit_train_batches", "global_step", "num_training_batches", "current_epoch", "should_train"],
+    [pytest.param(0.2, 0, 0, 0, False), pytest.param(0.5, 10, 2, 4, True)],
+)
+def test_disabled_training_for_insufficient_limit_train_batches(tmpdir, limit_train_batches, global_step,
+                                                                num_training_batches, current_epoch, should_train):
+    """
+    Verify when `limit_train_batches` is float & between [0.0, 0.1] and
```

Comment:
is `[0.0, 0.1]` correct in this case? I assume what we're actually checking for is if there are any training batches to train on in the first place (After accounting for limit_train_batches)

Reply:
Should be `1.0`, updated. We are testing if we have any training batches to train on with the `limited` training set by `limit_train_batches`. @SeanNaren Is this what you were suggesting?